pkg/trace/force_sampler.go

```diff
@@ -57,3 +58,53 @@ func sampleAt(ctx context.Context, rate uint) context.Context {
 	}
 	return ctx
 }
+
+type otelForceSampler struct {
+	sampleRate uint
+}
+
+func (s *otelForceSampler) Description() string {
+	return "Samples at the specified rate or forces sampling based on the `force_trace` attribute."
+}
+
+//nolint:gocritic // Why: Required to pass SamplingParameters as a copy
+func (s *otelForceSampler) ShouldSample(p sdktrace.SamplingParameters) sdktrace.SamplingResult {
+	psc := oteltrace.SpanContextFromContext(p.ParentContext)
+	sampler, err := sample.NewDeterministicSampler(s.sampleRate)
+	if err != nil {
+		panic(err)
+	}
+
+	var forceTrace bool
+	for _, a := range p.Attributes {
+		if string(a.Key) == fieldForceTrace {
+			forceTrace = a.Value.AsBool()
+		}
+	}
+
+	if forceTrace {
+		return sdktrace.SamplingResult{
+			Decision:   sdktrace.RecordAndSample,
+			Tracestate: psc.TraceState(),
+		}
```

Comment:
Honeycomb is pretty smart about sampling.

It knows what kind of dice we're rolling and can adjust counts accordingly.  If your services is sampling 1/100, it'll note this in the trace data, and in the UI it'll show all counts multiplied by 100 to make up for it.  If you decide to adjust your sampling rate to 1/200 the next day, it'll keep adjusting and the graphs in the UI will continue to look mostly the same.

If you have a *forced* trace, that means you have a sample rate of 100% for that particular trace.  Honeycomb knows how to account for that, too.  A forced trace will increment the counts by exactly one in the UI.  That way a couple of forced traces won't cause your graphs to look weird.

To support all this, Honeycomb samplers return not just a true/false, but also the sample rate used for that trace.  See [here](https://github.com/getoutreach/outreach-ruby-lib/blob/80dd73b3d1732cb363a963392ce3d07503714582/lib/outreach_trace/allow_force_sampler.rb#L27-L35).

Which brings me to the point of this long comment.  We don't seem to be returning anything about sampling rates here.  Does OTel not support any kind of concept like this?

Reply:
It looks like honeycomb supports this in OpenTelemetry through the use of attributes. Their documentation is a little confusing about how to accomplish this, but looks like it is possible. I will have to dig into it.

Reply:
Added the new attribute in accordance with Honeycomb's documentation https://docs.honeycomb.io/getting-data-in/opentelemetry/go/#sampling